jsonParser.go

```diff
@@ -130,39 +128,135 @@ type Link struct {
+	Issue           *string    `json:"issue"`
+	PublishedOnline *DateParts `json:"published-online"`
+	PublishedPrint  *DateParts `json:"published-print"`
 }
 type IssnType struct {
 	Value string `json:"value"`
 	Type  string `json:"type"`
 }
-type PublishedOther struct {
-	DateParts [][]int `json:"date-parts"`
-}
-type Published struct {
-	DateParts [][]int `json:"date-parts"`
-}
-
 type Relation struct {
 	Cites []any `json:"cites"`
 }
-
 type License struct {
 	URL            string  `json:"URL"`
 	Start          Indexed `json:"start"`
 	DelayInDays    int     `json:"delay-in-days"`
 	ContentVersion string  `json:"content-version"`
 }
+
+//
+func (pub *CrossRef) GetSimpleTitle() []string {
+	simpleTitle := pub.Title
+	if len(simpleTitle) == 0 {
+		return []string{""}
+	}
+	for pos, title := range simpleTitle {
+		title = strings.Replace(title, "\n", " ", -1)
+		title = strings.Replace(title, "( )+", " ", -1)
+		simpleTitle[pos] = strings.TrimSpace(title)
+	}
+	return simpleTitle
+}
+
+func (pub *CrossRef) GetSimpleDOI() string {
+	return pub.Doi
+}
+
+func (pub *CrossRef) GetSimpleAuthor() string {
+	if len(pub.Author) == 0 {
+		return ""
+	}
+	var simpleAuthor strings.Builder
+	for _, auth := range pub.Author {
+		simpleAuthor.WriteString(*auth.Family + " ")
+	}
+	return strings.TrimSpace(simpleAuthor.String())
+}
+
+func (pub *CrossRef) GetSimpleFirstAuthor() string {
+	if len(pub.Author) == 0 {
+		return ""
+	}
+	for _, auth := range pub.Author {
+		if *auth.Sequence == "first" && *auth.Family != "" {
+			return *auth.Family
+		}
+	}
+	// not sequence information apparently, so as fallback we use the first
+	// author in the author list
+	return *pub.Author[0].Family
+}
+
+func (pub *CrossRef) GetSimpleFirstPage() string {
+	sp := regexp.MustCompile(`,|-|\s`)
+	pagePieces := sp.Split(pub.Page, -1)
+	return pagePieces[0]
+}
+
+func (pub *CrossRef) GetSimpleJournal() string {
+	if len(pub.ContainerTitle) == 0 {
+		return ""
+	}
+	return pub.ContainerTitle[0]
+}
+func (pub *CrossRef) GetSimpleAbbreviatedJournal() string {
+	if len(*pub.ShortContainerTitle) == 0 {
+		return ""
+	}
+	return (*pub.ShortContainerTitle)[0]
+}
+
+func (pub *CrossRef) GetSimpleVolume() string {
+	return pub.Volume
+}
+
+func (pub *CrossRef) GetSimpleIssue() string {
+	return pub.Issue
+}
+
+// year is a date part (first one) in issued or created or published-online (we follow this order)
+func (pub *CrossRef) GetSimpleYear() int {
+	var year int
+	switch {
+	case pub.Issued.DateParts != nil:
+		year = pub.Issued.DateParts[0][0]
+	case pub.PublishedOnline != nil:
+		year = pub.PublishedOnline.DateParts[0][0]
+	case pub.PublishedPrint != nil:
+		year = pub.PublishedPrint.DateParts[0][0]
+	default:
+		// this is deposit date, normally we will never use it, but it will ensure
+		// that we always have a date as conservative fallback
+		year = pub.Created.DateParts[0][0]
+	}
+	return year
+}
+
+func BuildBibliographicField(pub *CrossRef) string {
```

Comment:
This whole function can be replaced with you adding all of the fields to a slice of strings and then filtering the slice to remove empty strings and then doing `strings.Join`. Then you don't have to mess with if-statements and builders

Reply:
but the filtering does not guarantee the expected result always.

Reply:
Ok, be sure to include one of those examples in the tests and we'll make sure to cover it when refactoring.